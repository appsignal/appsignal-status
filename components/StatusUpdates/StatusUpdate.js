import PropTypes from "prop-types";

import StatusIcon from "../StatusIcon/StatusIcon";

const StatusUpdate = ({ status, title, description, time }) => {
  return (
    <div className="flex bg-white shadow-sm rounded py-5 px-6">
      <div className="mr-4 my-0.5">
        <StatusIcon status={status} />
      </div>
      <div className="">
        <h3 className="c_h-heading mb-1">{title}</h3>
        {description && <p className="text-gray-700">{description}</p>}
        <p className="text-gray-600 text-ms mt-2">{time}</p>
      </div>
    </div>
  );
};

StatusUpdate.propTypes = {
  status: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string,
  time: PropTypes.string.isRequired,
};

export default StatusUpdate;
