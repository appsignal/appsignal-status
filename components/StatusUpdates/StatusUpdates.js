import PropTypes from "prop-types";
import StatusUpdate from "./StatusUpdate";

const statusUpdates = [
  {
    status: "success",
    title: "All systems are back up!",
    description: "",
    time: "15th Aug 08:11 CET",
  },
  {
    status: "error",
    title: "We’re experiencing some issues",
    description: "Here’s a message with some context.",
    time: "15th Aug 07:53 CET",
  },
  {
    status: "warning",
    title: "Planned maintainance on August 15th between 02:00pm and 03:00pm",
    description: "Here’s a message with some context.",
    time: "11th Aug 15:11 CET",
  },
];

const StatusUpdates = () => {
  return (
    <section className="mt-20 mb-16">
      <div className="container">
        <a name="status-updates" id="status-updates">
          <h2 className="c_h-heading c_h-heading--2xl mb-8 text-center">
            Status updates
          </h2>
        </a>
        <div className="max-w-lg mx-auto space-y-4">
          {statusUpdates.map((update, index) => (
            <StatusUpdate
              key={index}
              status={update.status}
              title={update.title}
              description={update.description}
              time={update.time}
            />
          ))}
        </div>
      </div>
    </section>
  );
};

StatusUpdates.propTypes = {
  statusUpdates: PropTypes.array.isRequired,
};

export default StatusUpdates;
