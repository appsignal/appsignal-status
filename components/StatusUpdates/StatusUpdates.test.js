import { render, screen } from "@testing-library/react";

import statusPageMock from "../../mocks/status_pages/appsignal.json";

import StatusUpdates from "./StatusUpdates";

const build = (props = {}) => {
  return render(<StatusUpdates updates={statusPageMock.updates} {...props} />);
};

describe("StatusUpdates", () => {
  test("it renders without errors", () => {
    const { container } = build();
    expect(container).toMatchSnapshot();
  });

  test("it renders multiple status updates", () => {
    build();

    expect(
      screen.getByText(statusPageMock.updates[0].title)
    ).toBeInTheDocument();
    expect(
      screen.getByText(statusPageMock.updates[1].title)
    ).toBeInTheDocument();
  });

  test("renders an empty state when there are no updates", () => {
    build({ updates: [] });

    expect(screen.getByText("No updates yet")).toBeInTheDocument();
  });
});
