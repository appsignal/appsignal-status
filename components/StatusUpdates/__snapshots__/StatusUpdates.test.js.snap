// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusUpdates it renders without errors 1`] = `
<div>
  <section
    class="mt-20 mb-16"
  >
    <div
      class="container"
    >
      <a
        class="no-underline"
        id="status-updates"
        name="status-updates"
      >
        <h2
          class="c_h-heading c_h-heading--2xl mb-8 text-center"
        >
          Status updates
        </h2>
      </a>
      <div
        class="max-w-lg mx-auto space-y-4"
      >
        <div
          class="flex bg-white shadow-sm rounded py-5 px-6"
          data-testid="statusUpdate"
        >
          <div
            class="mr-4 my-0.5"
          >
            <span
              class="flex items-center justify-center h-4 w-4 rounded-full bg-green-500 text-white"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-check fa-w-16 fa-fw "
                data-icon="check"
                data-prefix="fas"
                focusable="false"
                role="img"
                style="font-size: 8px;"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </div>
          <div
            class="min-w-0"
          >
            <h3
              class="c_h-heading mb-1"
            >
              All Fixed again
            </h3>
            <p
              class="text-gray-600 text-ms mt-2"
            >
              Sep. 1st 08:44
            </p>
          </div>
        </div>
        <div
          class="flex bg-white shadow-sm rounded py-5 px-6"
          data-testid="statusUpdate"
        >
          <div
            class="mr-4 my-0.5"
          >
            <span
              class="flex items-center justify-center h-4 w-4 rounded-full bg-red-500 text-white"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-exclamation fa-w-6 fa-fw "
                data-icon="exclamation"
                data-prefix="fas"
                focusable="false"
                role="img"
                style="font-size: 8px;"
                viewBox="0 0 192 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M176 432c0 44.112-35.888 80-80 80s-80-35.888-80-80 35.888-80 80-80 80 35.888 80 80zM25.26 25.199l13.6 272C39.499 309.972 50.041 320 62.83 320h66.34c12.789 0 23.331-10.028 23.97-22.801l13.6-272C167.425 11.49 156.496 0 142.77 0H49.23C35.504 0 24.575 11.49 25.26 25.199z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </div>
          <div
            class="min-w-0"
          >
            <h3
              class="c_h-heading mb-1"
            >
              Some update
            </h3>
            <p
              class="text-gray-700 break-words"
            >
              Some description
            </p>
            <p
              class="text-gray-600 text-ms mt-2"
            >
              Aug. 30th 08:43
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
